ddtrace/bootstrap/sitecustomize.py

```diff
@@ -162,6 +91,52 @@ def update_patched_modules():
         EXTRA_PATCHED_MODULES[module] = asbool(should_patch)
 
 
+if PY2:
+    _unloaded_modules = []
+
+
+def is_module_installed(module_name):
+    return find_loader(module_name) is not None
+
+
+def cleanup_loaded_modules():
+    MODULES_REQUIRING_CLEANUP = ("gevent",)
+    do_cleanup = os.getenv("DD_UNLOAD_MODULES_FROM_SITECUSTOMIZE", default="auto").lower()
+    if do_cleanup == "auto":
+        do_cleanup = any(is_module_installed(m) for m in MODULES_REQUIRING_CLEANUP)
+
+    if not asbool(do_cleanup):
+        return
+
+    # Unload all the modules that we have imported, except for the ddtrace one.
+    # NB: this means that every `import threading` anywhere in `ddtrace/` code
+    # uses a copy of that module that is distinct from the copy that user code
+    # gets when it does `import threading`. The same applies to every module
+    # not in `KEEP_MODULES`.
+    KEEP_MODULES = frozenset(["atexit", "ddtrace", "asyncio", "concurrent", "typing", "logging", "attr"])
```

Comment:
nit: I don't think we get much value from defining a frozen set here. Should we define this constant on the module level?

Reply:
@mabdinur Can you help me understand the downside of using a frozenset in this way?

Reply:
This constant is only used by this helper function, so I don't see much value in moving it to a global level.